ledger/participant-integration-api/src/test/suite/scala/platform/indexer/RecoveringIndexerSpec.scala

```diff
@@ -101,43 +105,43 @@ final class RecoveringIndexerSpec
           Level.INFO -> "Stopped Indexer Server",
         )
         testIndexer.openSubscriptions shouldBe mutable.Set.empty
+        healthReporter.currentHealth() shouldBe HealthStatus.healthy
       }
     }
 
     "wait until the subscription completes" in {
       val recoveringIndexer =
-        new RecoveringIndexer(actorSystem.scheduler, actorSystem.dispatcher, 10.millis)
+        RecoveringIndexer(actorSystem.scheduler, actorSystem.dispatcher, 10.millis)
       val testIndexer = new TestIndexer(
         SubscribeResult("A", SuccessfullyCompletes, 100.millis, 10.millis)
       )
 
       val resource = recoveringIndexer.start(() => testIndexer.subscribe())
-      resource.asFuture
-        .map { complete =>
-          // at this point the read log should at least contain logs informing about a successful indexer server startup
+      for {
+        (healthReporter, complete) <- resource.asFuture
+        _ <- Future {
           readLog().take(2) should contain theSameElementsInOrderAs Seq(
             Level.INFO -> "Starting Indexer Server",
             Level.INFO -> "Started Indexer Server",
           )
-          complete
-        }
-        .flatten
-        .transformWith(finallyRelease(resource))
-        .map { _ =>
-          readLog() should contain theSameElementsInOrderAs Seq(
-            Level.INFO -> "Starting Indexer Server",
-            Level.INFO -> "Started Indexer Server",
-            Level.INFO -> "Successfully finished processing state updates",
-            Level.INFO -> "Stopping Indexer Server",
-            Level.INFO -> "Stopped Indexer Server",
-          )
-          testIndexer.openSubscriptions shouldBe mutable.Set.empty
         }
+        _ <- complete.transformWith(finallyRelease(resource))
+      } yield {
+        readLog() should contain theSameElementsInOrderAs Seq(
+          Level.INFO -> "Starting Indexer Server",
+          Level.INFO -> "Started Indexer Server",
+          Level.INFO -> "Successfully finished processing state updates",
+          Level.INFO -> "Stopping Indexer Server",
+          Level.INFO -> "Stopped Indexer Server",
+        )
+        testIndexer.openSubscriptions shouldBe mutable.Set.empty
+        healthReporter.currentHealth() shouldBe HealthStatus.healthy
```

Comment:
Does this imply that a stopped indexer server is still healthy?

Reply:
If it shut down gracefully, yes.. Do you think it should be non-healthy?

Reply:
What does the client expect? E.g., can it expect the service that propagates this health status to be still working or not?

Reply:
I'm not sold on either message:
If the service is stopped, it's weird to argue that it is healthy. On the other hand, it is weird to also say that it is unhealthy.

Reply:
Ah, I think this settles it. This is what grpcurl prints for unhealthy/healthy:
```
{
  "status": "NOT_SERVING"
}
{
  "status": "SERVING"
}
```
So we will go with: unhealthy if the service has stopped.